src/spatialdata/_core/spatialdata.py

```diff
@@ -962,195 +908,564 @@ def transform_to_coordinate_system(
+        saving_an_element: bool = False,
     ) -> None:
-        from spatialdata._io import write_image, write_labels, write_points, write_shapes, write_table
-        from spatialdata._io._utils import get_dask_backing_files
+        from spatialdata._io._utils import _backed_elements_contained_in_path, _is_subfolder
 
-        """Write the SpatialData object to Zarr."""
         if isinstance(file_path, str):
             file_path = Path(file_path)
-        assert isinstance(file_path, Path)
-
-        if self.is_backed() and str(self.path) != str(file_path):
-            logger.info(f"The Zarr file used for backing will now change from {self.path} to {file_path}")
 
-        # old code to support overwriting the backing file
-        # target_path = None
-        # tmp_zarr_file = None
+        if not isinstance(file_path, Path):
+            raise ValueError(f"file_path must be a string or a Path object, type(file_path) = {type(file_path)}.")
 
         if os.path.exists(file_path):
             if parse_url(file_path, mode="r") is None:
                 raise ValueError(
-                    "The target file path specified already exists, and it has been detected to not be "
-                    "a Zarr store. Overwriting non-Zarr stores is not supported to prevent accidental "
-                    "data loss."
+                    "The target file path specified already exists, and it has been detected to not be a Zarr store. "
+                    "Overwriting non-Zarr stores is not supported to prevent accidental data loss."
                 )
             if not overwrite:
-                raise ValueError("The Zarr store already exists. Use `overwrite=True` to overwrite the store.")
-            if self.is_backed() and str(self.path) == str(file_path):
                 raise ValueError(
-                    "The file path specified is the same as the one used for backing. "
-                    "Overwriting the backing file is not supported to prevent accidental data loss."
-                    "We are discussing how to support this use case in the future, if you would like us to "
-                    "support it please leave a comment on https://github.com/scverse/spatialdata/pull/138"
+                    "The Zarr store already exists. Use `overwrite=True` to try overwriting the store."
+                    "Please note that only Zarr stores not currently in used by the current SpatialData object can be "
+                    "overwritten."
                 )
-            if any(Path(fp).resolve().is_relative_to(file_path.resolve()) for fp in get_dask_backing_files(self)):
+            ERROR_MSG = (
+                "Cannot overwrite. The target path of the write operation is in use. Please save the data to a "
+                "different location. "
+            )
+            WORKAROUND = (
+                "\nWorkaround: please see discussion here https://github.com/scverse/spatialdata/discussions/520."
```

Comment:
this link is dead, I think because of the .

Reply:
Fixed it now.